File: Python/Stack.py
# Implementation of stack using Linked list

class Node:
    def __init__(self, data):
        self.next = None
        self.data = data


class Stack:
    def __init__(self):
        self.head = None

    # For adding data
    def push(self, data):
        node = Node(data)
        node.next, self.head = self.head, node

    # For fetching the data
    def pop(self):
        if not self.head:
            raise Exception("Popping from the empty stack []")
        storage = self.head.data
        self.head = self.head.next
        return storage

    # For displaying stack
    def print_stack(self):
        storage, current = [], self.head
        while current:
            storage.append(current.data)
            current = current.next
        print(storage)


if __name__ == '__main__':
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.print_stack()
    print(stack.pop())
    stack.print_stack()

File: C++/Stack.cpp
//
// Created by Anubhav Shukla on 23-08-2022.
//

#include<iostream>
#include<vector>
using namespace std;

class Node{
public:
    int data;
    Node* next;  // address for the Node
    explicit Node(int data){
        this -> data = data;
        this -> next = nullptr;
    }
};

class Stack{
public:
    Node* head;
    Stack(){
        this -> head = nullptr;
    }

    // For adding data
    void push(int data){
        Node* newNode = new Node(data);
        newNode -> next = head;
        head = newNode;
    }

    // For popping
    int pop(){
        if(!head){
            throw invalid_argument("Popping from the empty stack []");
        }
        int storage = head -> data;
        head = head -> next;
        return storage;

    }

    // For printing
    void print() const{
        Node* current = head;
        vector<int> storage;
        while(current){
            storage.push_back(current->data);
            current = current -> next;
        }
        cout << '[';
        for(int i = 0; i < storage.size(); i++){
            if(i == storage.size() - 1){
                cout << storage[i];
            }
            else{
                cout << storage[i] << ", ";
            }
        }
        cout << ']' << endl;
    }
};

int main(){
    auto* stack = new Stack();
    stack->push(10);
    stack -> push(20);
    stack -> print();
    cout << stack -> pop() << endl;
    stack -> print();
    return 0;
}